OptimizelySDK.Tests/NotificationTests/NotificationCenterTests.cs

```diff
@@ -0,0 +1,251 @@
+            NotificationCenter.ClearAllNotifications();
+            NotificationCenter.ClearNotifications(NotificationTypeTrack);
+        }
+
+        [Test]
+        public void TestClearAllNotifications()
+        {
+            // Add decision notifications.
+            NotificationCenter.AddNotification(NotificationTypeActivate, TestNotificationCallbacks.TestActivateCallback);
+            NotificationCenter.AddNotification(NotificationTypeActivate, TestNotificationCallbacks.TestAnotherActivateCallback);
+
+            // Add track notification.
+            NotificationCenter.AddNotification(NotificationTypeTrack, TestNotificationCallbacks.TestTrackCallback);
+
+            // Verify that callbacks added successfully.
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that ClearAllNotifications remove all the callbacks.
+            NotificationCenter.ClearAllNotifications();
+            Assert.AreEqual(0, NotificationCenter.NotificationsCount);
+
+            // Verify that ClearAllNotifications does not break on calling twice or after ClearNotifications.
+            NotificationCenter.ClearNotifications(NotificationTypeActivate);
+            NotificationCenter.ClearAllNotifications();
+            NotificationCenter.ClearAllNotifications();
+        }
+
+        [Test]
+        public void TestSendNotifications()
+        {
+            var config = ProjectConfig.Create(TestData.Datafile, LoggerMock.Object, new NoOpErrorHandler());
+
+            // Mocking notification callbacks.
+            var notificationCallbackMock = new Mock<TestNotificationCallbacks>();
+
+            notificationCallbackMock.Setup(nc => nc.TestActivateCallback(It.IsAny<Experiment>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()));
+
+            notificationCallbackMock.Setup(nc => nc.TestAnotherActivateCallback(It.IsAny<Experiment>(), 
+                It.IsAny<string>(), It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()));
+
+            notificationCallbackMock.Setup(nc => nc.TestFeatureRolloutCallback(It.IsAny<string>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<Audience[]>()));
+
+            // Adding decision notifications.
+            NotificationCenter.AddNotification(NotificationTypeActivate, notificationCallbackMock.Object.TestActivateCallback);
+            NotificationCenter.AddNotification(NotificationTypeActivate, notificationCallbackMock.Object.TestAnotherActivateCallback);
+
+
+            // Adding track notifications.
+            NotificationCenter.AddNotification(NotificationTypeTrack, notificationCallbackMock.Object.TestTrackCallback);
+            
+            // Fire decision type notifications.
+            NotificationCenter.SendNotifications(NotificationTypeActivate, config.GetExperimentFromKey("test_experiment"), 
```

Comment:
The LogEvent would never be null.  So maybe mock it?

Reply:
mocked it. Please check. https://github.com/optimizely/csharp-sdk/pull/35/commits/0e80f49306aecfd798543bbf0817733f487b55bc